tests/recipe_training_tests/coded_qat_launch_test.py

```diff
@@ -0,0 +1,39 @@
+import unittest
+
+from super_gradients import QATTrainer
+from super_gradients.training.dataloaders.dataloaders import classification_test_dataloader
+from super_gradients.training.metrics import Accuracy, Top5
+from super_gradients.training.models import ResNet18
+
+
+class CodedQATLuanchTest(unittest.TestCase):
+    def test_qat_launch(self):
+        trainer = QATTrainer("test_launch_qat_with_minimal_changes")
+        net = ResNet18(num_classes=5, arch_params={})
+        train_params = {
+            "max_epochs": 2,
+            "lr_updates": [1],
+            "lr_decay_factor": 0.1,
+            "lr_mode": "step",
+            "lr_warmup_epochs": 0,
+            "initial_lr": 0.1,
+            "loss": "cross_entropy",
+            "optimizer": "SGD",
+            "criterion_params": {},
+            "optimizer_params": {"weight_decay": 1e-4, "momentum": 0.9},
+            "train_metrics_list": [Accuracy(), Top5()],
+            "valid_metrics_list": [Accuracy(), Top5()],
+            "metric_to_watch": "Accuracy",
+            "greater_metric_to_watch_is_better": True,
+        }
+        trainer.train(
+            model=net,
+            training_params=train_params,
+            train_loader=classification_test_dataloader(batch_size=10),
+            valid_loader=classification_test_dataloader(batch_size=10),
+        )
+        trainer.quantize(calib_dataloader=classification_test_dataloader(batch_size=10))
```

Comment:
`trainer.train()` does QAT. This is very clear and intuitive.
`trainer.quantize()` does PTQ + optionally QAT. And by default it does both. I think this can be missleading, people might want to run `trainer.train()` and then `trainer.quantize()` and not think about setting `ptq_only=True`.

Maybe we should drop the name `trainer.quantize()` and instead have `trainer.post_training_quantization()`
If we still want the option to run to do both qat and ptq together, we could have something like `trainer.full_quantization()`

Reply:
Agree to @Louis-Dupont. I haven't been into QAT so all this is relatively new to me.
So as a newcomer, I'd expect `trainer.quantize` to do a quantization of the already trained model. The method name does not give any hints that training may take place. Perhaps for sake of continence we can have:

```python
def train_and_quantize(...):
   self.train(...)
   self.quantize(...)
```

Second, I believe that both Trainer and QATTrainer should both have `quantize` method.
Why? 
Regardless whether we use QAT or not, we should be able to use PTQ. For that reason, having a `quantize` method does make sense to me.
However, to get the maximum accuracy, one may employ QATTrainer which would do it's QAT-specific magic:

So user flow would be:
1) Do I want to get quantized model?
[NO] -> Use `Trainer.train()`
[YES] -> Use `Trainer.train_and_quantize()`
2) Do I want to train model with quantization-aware tricks?
[NO] -> Use `Trainer.train_and_quantize()`
[YES] -> Use `QATTrainer.train_and_quantize()`